core/src/main/java/org/geysermc/geyser/registry/type/BlockMappings.java

```diff
@@ -75,6 +76,10 @@ public BlockDefinition getItemFrame(NbtMap tag) {
     }
 
     public boolean isItemFrame(BlockDefinition definition) {
-        return this.itemFrames.containsKey(definition.getRuntimeId());
+        if (definition instanceof SimpleBlockDefinition def) {
```

Comment:
Not sure if I like this. Will merge without it being changed but BlockDefinition should always be a certain type.

Reply:
> BlockDefinition should always be a certain type.

In regards to the packet translator that uses this?

Reply:
Checking for instanceof here implies that BlockDefinition might not be SimpleBlockDefinition, when (if I am correct) it'll always be SimpleBlockDefinition within Geyser.

Reply:
Ah - I guess we could add a warning if it isn't?